File: javascript/challenge-2025-09-25.js

function compareNumbers(a, b) {
  return a - b;
}

function secondLargest(arr) {
  let result = Infinity;
  console.log("arr (1):", arr);
  arr.sort(compareNumbers).reverse();
  console.log("arr (2):", arr);

  let assignments = 0;
  let i = 0;
  while (assignments < 2) {
    if (arr[i] < result) {
      result = arr[i];
      assignments += 1;
    }
    i += 1;
  }
  console.log("assignments:", assignments);
  console.log("result:", result);
  return result;
}

File: python/challenge-2025-09-25.py

def second_largest(arr):
    result = float("inf")
    arr.sort(reverse=True)

    assignments = 0
    i = 0
    while assignments < 2:
        if arr[i] < result:
            result = arr[i]
            assignments += 1
        i += 1
    return result
